tools/clang/lib/SPIRV/DeclResultIdMapper.cpp

```diff
@@ -113,6 +95,34 @@ inline QualType getTypeOrFnRetType(const DeclaratorDecl *decl) {
 }
 } // anonymous namespace
 
+std::string StageVar::getSemanticStr() const {
+  // A special case for zero index, which is equivalent to no index.
+  // Use what is in the source code.
+  // TODO: this looks like a hack to make the current tests happy.
```

Comment:
It's not necessarily a hack IMO... for normal cases (most of the time), it is more readable for users to have the `in_var_SEMANTIC` stage variable rather than `in_var_SEMANTIC0`.
Just a thought

Reply:
Yep. That's what I'm thinking. So didn't just remove the special case handling code. Put a TODO here as a reminder.